tests/src/Kernel/ResolverRegistryTest.php

```diff
@@ -0,0 +1,125 @@
+<?php
+
+namespace Drupal\Tests\graphql\Kernel;
+
+use Drupal\graphql\GraphQL\ResolverRegistry;
+
+/**
+ * Tests that the resolver registry behaves correctly.
+ *
+ * @coversDefaultClass \Drupal\graphql\GraphQL\ResolverRegistry
+ *
+ * @group graphql
+ */
+class ResolverRegistryTest extends GraphQLTestBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setUp(): void {
+    parent::setUp();
+
+    $schema = <<<GQL
+      type Query {
+        transportation: Vehicle
+      }
+
+      interface Vehicle {
+        model: String
+      }
+
+      interface Car implements Vehicle {
+        model: String
+      }
+
+      type Tesla implements Car {
```

Comment:
not super happy about using product names of our billionaire friend that does not like labor laws. I would suggest "Cabrio", but not a blocker for me.

Reply:
Very fair point! I was lacking some inspiration, happy to give this a different name.